From 1df0e4b2dbf4f42e11ee1844237f81dbe380a578 Mon Sep 17 00:00:00 2001
From: Bryan Brattlof <bb@ti.com>
Date: Mon, 10 Feb 2025 22:03:40 -0600
Subject: [PATCH 65/68] PENDING: arm64: dts: ti: k3-am62: add SCMI nodes

The AM62Lx uses SCMI to communicate to TF-A which handles PLL and LPSC
change requests for Linux. Add in the shared memory and nodes for the
SCMI subsystem for the other nodes.

Signed-off-by: Bryan Brattlof <bb@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am62l-main.dtsi |  6 ++++++
 arch/arm64/boot/dts/ti/k3-am62l.dtsi      | 25 +++++++++++++++++++++++
 2 files changed, 31 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-am62l-main.dtsi b/arch/arm64/boot/dts/ti/k3-am62l-main.dtsi
index db94ba496bfb..9eca95eae04b 100644
--- a/arch/arm64/boot/dts/ti/k3-am62l-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62l-main.dtsi
@@ -60,5 +60,11 @@ oc_sram: sram@70800000 {
 		ranges = <0x0 0x00 0x70800000 0x10000>;
 		#address-cells = <1>;
 		#size-cells = <1>;
+
+		scmi_shmem: sram@0 {
+			compatible = "arm,scmi-shmem";
+			reg = <0x0 0x100>;
+			bootph-all;
+		};
 	};
 };
diff --git a/arch/arm64/boot/dts/ti/k3-am62l.dtsi b/arch/arm64/boot/dts/ti/k3-am62l.dtsi
index 5a77f32d1eb1..d60366b8d112 100644
--- a/arch/arm64/boot/dts/ti/k3-am62l.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62l.dtsi
@@ -24,6 +24,31 @@ optee {
 			compatible = "linaro,optee-tz";
 			method = "smc";
 		};
+
+		psci: psci {
+			compatible = "arm,psci-1.0";
+			method = "smc";
+		};
+
+		scmi: scmi {
+			compatible = "arm,scmi-smc";
+			arm,smc-id = <0x82004000>;
+			shmem = <&scmi_shmem>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			scmi_clk: protocol@14 {
+				reg = <0x14>;
+				#clock-cells = <1>;
+				bootph-all;
+			};
+
+			scmi_pds: protocol@11 {
+				reg = <0x11>;
+				#power-domain-cells = <1>;
+				bootph-all;
+			};
+		};
 	};
 
 	a53_timer0: timer-cl0-cpu0 {
-- 
2.47.2

